[Issue]: compile errors with libc++

### Problem Description

Hi, there are 2 issues when compiled with libc++ (the default C++ Standard Library implementation for many major platforms, including Apple’s macOS, iOS, watchOS, and tvOS, Google Search, the Android operating system, and FreeBSD[1]). I checked both and they seem to be related to composable_kernel, not libc++.

First issue is attempt to call `std::numeric_limits<ranges::range_value_t<_Float16>>::min()`. `_Float16` is extension of libstdc++, it does not exist in C++ standard[2]. Luckily, as I see there is `NumericLimits` class in composable_kernel, which does everything needed.
<details><summary>Full error log</summary>
<p>

```c++
FAILED: profiler/src/CMakeFiles/ckProfiler.dir/profile_gemm_splitk.cpp.o 
/usr/bin/hipcc -DCK_ENABLE_BF16 -DCK_ENABLE_BF8 -DCK_ENABLE_FP16 -DCK_ENABLE_FP32 -DCK_ENABLE_FP64 -DCK_ENABLE_FP8 -DCK_ENABLE_INT8 -DUSE_PROF_API=1 -D__HIP_PLATFORM_AMD__=1 -D__HIP_PLATFORM_HCC__=1 -I/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/profiler/include -I/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/library/include -I/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/include -I/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1_build/include  -DNDEBUG -std=c++17   -Wall -Wextra -Wcomment -Wendif-labels -Wformat -Winit-self -Wreturn-type -Wsequence-point -Wswitch -Wtrigraphs -Wundef -Wuninitialized -Wunreachable-code -Wunused -Wno-reserved-identifier -Wno-option-ignored -Wsign-compare -Wno-extra-semi-stmt -Wno-unused-template -Wno-missing-field-initializers -Wno-deprecated-declarations -Wall -Wextra -Wcomment -Wendif-labels -Wformat -Winit-self -Wreturn-type -Wsequence-point -Wswitch -Wtrigraphs -Wundef -Wuninitialized -Wunreachable-code -Wunused -Wno-reserved-identifier -Wno-option-ignored -Wsign-compare -Wno-extra-semi-stmt -Wno-unused-template -Weverything -Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-conversion -Wno-double-promotion -Wno-exit-time-destructors -Wno-extra-semi -Wno-float-conversion -Wno-gnu-anonymous-struct -Wno-gnu-zero-variadic-macro-arguments -Wno-missing-prototypes -Wno-nested-anon-types -Wno-padded -Wno-return-std-move-in-c++11 -Wno-shorten-64-to-32 -Wno-sign-conversion -Wno-unknown-warning-option -Wno-unused-command-line-argument -Wno-weak-vtables -Wno-covered-switch-default -Wno-unsafe-buffer-usage -Wno-bit-int-extension -Wno-pass-failed -Wno-switch-default -fno-offload-uniform-block -Wno-global-constructors -x hip --offload-arch=gfx1030 -MD -MT profiler/src/CMakeFiles/ckProfiler.dir/profile_gemm_splitk.cpp.o -MF profiler/src/CMakeFiles/ckProfiler.dir/profile_gemm_splitk.cpp.o.d -o profiler/src/CMakeFiles/ckProfiler.dir/profile_gemm_splitk.cpp.o -c /var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/profiler/src/profile_gemm_splitk.cpp
In file included from <built-in>:1:
In file included from /usr/lib/llvm/18/bin/../../../../lib/clang/18/include/__clang_hip_runtime_wrapper.h:111:
In file included from /usr/lib/llvm/18/bin/../../../../lib/clang/18/include/cuda_wrappers/cmath:27:
In file included from /usr/include/c++/v1/cmath:316:
/usr/include/c++/v1/limits:181:57: error: invalid operands to binary expression ('_Float16' and 'int')
  181 |   static _LIBCPP_CONSTEXPR const _Tp value = _Tp(_Tp(1) << __digits);
      |                                                  ~~~~~~ ^  ~~~~~~~~
/usr/include/c++/v1/limits:200:100: note: in instantiation of static data member 'std::__libcpp_compute_min<_Float16, 15, true>::value' requested here
  200 |   static _LIBCPP_CONSTEXPR const type __min       = __libcpp_compute_min<type, digits, is_signed>::value;
      |                                                                                                    ^
/usr/include/c++/v1/limits:202:80: note: in instantiation of static data member 'std::__libcpp_numeric_limits<_Float16>::__min' requested here
  202 |   _LIBCPP_HIDE_FROM_ABI static _LIBCPP_CONSTEXPR type min() _NOEXCEPT { return __min; }
      |                                                                                ^
/usr/include/c++/v1/limits:445:88: note: in instantiation of member function 'std::__libcpp_numeric_limits<_Float16>::min' requested here
  445 |   _LIBCPP_HIDE_FROM_ABI static _LIBCPP_CONSTEXPR type min() _NOEXCEPT { return __base::min(); }
      |                                                                                        ^
/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/library/include/ck/library/utility/check_err.hpp:149:73: note: in instantiation of member function 'std::numeric_limits<_Float16>::min' requested here
  149 |     double max_err = std::numeric_limits<ranges::range_value_t<Range>>::min();
      |                                                                         ^
/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/profiler/include/profiler/profile_gemm_splitk_impl.hpp:189:46: note: in instantiation of function template specialization 'ck::utils::check_err<Tensor<_Float16>, Tensor<_Float16>>' requested here
  189 |                     pass = pass & ck::utils::check_err(c_m_n_device_result, c_m_n_host_result);
      |                                              ^
/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/profiler/src/profile_gemm_splitk.cpp:113:35: note: in instantiation of function template specialization 'ck::profiler::profile_gemm_splitk_impl<_Float16, _Float16, float, _Float16, ck::tensor_layout::gemm::RowMajor, ck::tensor_layout::gemm::RowMajor, ck::tensor_layout::gemm::RowMajor, _Float16>' requested here
  113 |         bool pass = ck::profiler::profile_gemm_splitk_impl<ADataType,
      |                                   ^
/usr/include/c++/v1/limits:200:39: note: const variable cannot be emitted on device side due to dynamic initialization
  200 |   static _LIBCPP_CONSTEXPR const type __min       = __libcpp_compute_min<type, digits, is_signed>::value;
      |                   
```

</p>
</details> 

Second issue with `call to 'check_err' is ambiguous`: there are 2 candidates. It happens because composable_kernel relies on idea that `f8_t` (defined as `_BitInt(8)`) does not pass `is_integral` trait. However, due to inconsistent implementation, libc++ treats `_BitInt(8)` as integral. This was reported as https://github.com/llvm/llvm-project/issues/102767. But the fix is just one line, so I'd like to see it too.

<details><summary>Full error log</summary>
<p>

```c++
FAILED: profiler/src/CMakeFiles/ckProfiler.dir/profile_gemm.cpp.o 
/usr/bin/hipcc -DCK_ENABLE_BF16 -DCK_ENABLE_BF8 -DCK_ENABLE_FP16 -DCK_ENABLE_FP32 -DCK_ENABLE_FP64 -DCK_ENABLE_FP8 -DCK_ENABLE_INT8 -DUSE_PROF_API=1 -D__HIP_PLATFORM_AMD__=1 -D__HIP_PLATFORM_HCC__=1 -I/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/profiler/include -I/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/library/include -I/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/include -I/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1_build/include  -DNDEBUG -std=c++17   -Wall -Wextra -Wcomment -Wendif-labels -Wformat -Winit-self -Wreturn-type -Wsequence-point -Wswitch -Wtrigraphs -Wundef -Wuninitialized -Wunreachable-code -Wunused -Wno-reserved-identifier -Wno-option-ignored -Wsign-compare -Wno-extra-semi-stmt -Wno-unused-template -Wno-missing-field-initializers -Wno-deprecated-declarations -Wall -Wextra -Wcomment -Wendif-labels -Wformat -Winit-self -Wreturn-type -Wsequence-point -Wswitch -Wtrigraphs -Wundef -Wuninitialized -Wunreachable-code -Wunused -Wno-reserved-identifier -Wno-option-ignored -Wsign-compare -Wno-extra-semi-stmt -Wno-unused-template -Weverything -Wno-c++98-compat -Wno-c++98-compat-pedantic -Wno-conversion -Wno-double-promotion -Wno-exit-time-destructors -Wno-extra-semi -Wno-float-conversion -Wno-gnu-anonymous-struct -Wno-gnu-zero-variadic-macro-arguments -Wno-missing-prototypes -Wno-nested-anon-types -Wno-padded -Wno-return-std-move-in-c++11 -Wno-shorten-64-to-32 -Wno-sign-conversion -Wno-unknown-warning-option -Wno-unused-command-line-argument -Wno-weak-vtables -Wno-covered-switch-default -Wno-unsafe-buffer-usage -Wno-bit-int-extension -Wno-pass-failed -Wno-switch-default -fno-offload-uniform-block -Wno-global-constructors -x hip --offload-arch=gfx1030 -MD -MT profiler/src/CMakeFiles/ckProfiler.dir/profile_gemm.cpp.o -MF profiler/src/CMakeFiles/ckProfiler.dir/profile_gemm.cpp.o.d -o profiler/src/CMakeFiles/ckProfiler.dir/profile_gemm.cpp.o -c /var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/profiler/src/profile_gemm.cpp
In file included from /var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/profiler/src/profile_gemm.cpp:9:
/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/profiler/include/profiler/profile_gemm_impl.hpp:195:31: error: call to 'check_err' is ambiguous
  195 |                 pass = pass & ck::utils::check_err(c_m_n_device_result, c_m_n_host_result);
      |                               ^~~~~~~~~~~~~~~~~~~~
/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/profiler/src/profile_gemm.cpp:118:27: note: in instantiation of function template specialization 'ck::profiler::profile_gemm_impl<ck::tensor_layout::gemm::RowMajor, ck::tensor_layout::gemm::RowMajor, ck::tensor_layout::gemm::RowMajor, _BitInt(8), _BitInt(8), float, _BitInt(8)>' requested here
  118 |             ck::profiler::profile_gemm_impl<ALayout,
      |                           ^
/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/library/include/ck/library/utility/check_err.hpp:187:1: note: candidate function [with Range = Tensor<_BitInt(8)>, RefRange = Tensor<_BitInt(8)>]
  187 | check_err(const Range& out,
      | ^
/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/library/include/ck/library/utility/check_err.hpp:237:1: note: candidate function [with Range = Tensor<_BitInt(8)>, RefRange = Tensor<_BitInt(8)>]
  237 | check_err(const Range& out,
      | ^
In file included from /var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/profiler/src/profile_gemm.cpp:9:
/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/profiler/include/profiler/profile_gemm_impl.hpp:195:31: error: call to 'check_err' is ambiguous
  195 |                 pass = pass & ck::utils::check_err(c_m_n_device_result, c_m_n_host_result);
      |                               ^~~~~~~~~~~~~~~~~~~~
/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/profiler/src/profile_gemm.cpp:118:27: note: in instantiation of function template specialization 'ck::profiler::profile_gemm_impl<ck::tensor_layout::gemm::RowMajor, ck::tensor_layout::gemm::ColumnMajor, ck::tensor_layout::gemm::RowMajor, _BitInt(8), _BitInt(8), float, _BitInt(8)>' requested here
  118 |             ck::profiler::profile_gemm_impl<ALayout,
      |                           ^
/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/library/include/ck/library/utility/check_err.hpp:187:1: note: candidate function [with Range = Tensor<_BitInt(8)>, RefRange = Tensor<_BitInt(8)>]
  187 | check_err(const Range& out,
      | ^
/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/library/include/ck/library/utility/check_err.hpp:237:1: note: candidate function [with Range = Tensor<_BitInt(8)>, RefRange = Tensor<_BitInt(8)>]
  237 | check_err(const Range& out,
      | ^
In file included from /var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/profiler/src/profile_gemm.cpp:9:
/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/profiler/include/profiler/profile_gemm_impl.hpp:195:31: error: call to 'check_err' is ambiguous
  195 |                 pass = pass & ck::utils::check_err(c_m_n_device_result, c_m_n_host_result);
      |                               ^~~~~~~~~~~~~~~~~~~~
/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/profiler/src/profile_gemm.cpp:118:27: note: in instantiation of function template specialization 'ck::profiler::profile_gemm_impl<ck::tensor_layout::gemm::ColumnMajor, ck::tensor_layout::gemm::RowMajor, ck::tensor_layout::gemm::RowMajor, _BitInt(8), _BitInt(8), float, _BitInt(8)>' requested here
  118 |             ck::profiler::profile_gemm_impl<ALayout,
      |                           ^
/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/library/include/ck/library/utility/check_err.hpp:187:1: note: candidate function [with Range = Tensor<_BitInt(8)>, RefRange = Tensor<_BitInt(8)>]
  187 | check_err(const Range& out,
      | ^
/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/library/include/ck/library/utility/check_err.hpp:237:1: note: candidate function [with Range = Tensor<_BitInt(8)>, RefRange = Tensor<_BitInt(8)>]
  237 | check_err(const Range& out,
      | ^
In file included from /var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/profiler/src/profile_gemm.cpp:9:
/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/profiler/include/profiler/profile_gemm_impl.hpp:195:31: error: call to 'check_err' is ambiguous
  195 |                 pass = pass & ck::utils::check_err(c_m_n_device_result, c_m_n_host_result);
      |                               ^~~~~~~~~~~~~~~~~~~~
/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/profiler/src/profile_gemm.cpp:118:27: note: in instantiation of function template specialization 'ck::profiler::profile_gemm_impl<ck::tensor_layout::gemm::ColumnMajor, ck::tensor_layout::gemm::ColumnMajor, ck::tensor_layout::gemm::RowMajor, _BitInt(8), _BitInt(8), float, _BitInt(8)>' requested here
  118 |             ck::profiler::profile_gemm_impl<ALayout,
      |                           ^
/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/library/include/ck/library/utility/check_err.hpp:187:1: note: candidate function [with Range = Tensor<_BitInt(8)>, RefRange = Tensor<_BitInt(8)>]
  187 | check_err(const Range& out,
      | ^
/var/tmp/portage/sci-libs/composable-kernel-6.1.1/work/composable_kernel-rocm-6.1.1/library/include/ck/library/utility/check_err.hpp:237:1: note: candidate function [with Range = Tensor<_BitInt(8)>, RefRange = Tensor<_BitInt(8)>]
  237 | check_err(const Range& out,
      | ^
```

</p>
</details> 

I'll provide the pull-request with a fix, could you review it? Thanks!

[1] https://libcxx.llvm.org/#current-status
[2] https://en.cppreference.com/w/cpp/types/numeric_limits

### Operating System

Gentoo / llvm profile

### ROCm Version

ROCm 6.2.0

--- a/library/include/ck/library/utility/check_err.hpp
+++ b/library/include/ck/library/utility/check_err.hpp
@@ -146,7 +146,7 @@ check_err(const Range& out,
     bool res{true};
     int err_count  = 0;
     double err     = 0;
-    double max_err = std::numeric_limits<ranges::range_value_t<Range>>::min();
+    double max_err = NumericLimits<ranges::range_value_t<Range>>::Min();
     for(std::size_t i = 0; i < ref.size(); ++i)
     {
         const double o = type_convert<float>(*std::next(std::begin(out), i));
@@ -178,7 +178,9 @@ check_err(const Range& out,
 template <typename Range, typename RefRange>
 std::enable_if_t<(std::is_same_v<ranges::range_value_t<Range>, ranges::range_value_t<RefRange>> &&
                   std::is_integral_v<ranges::range_value_t<Range>> &&
-                  !std::is_same_v<ranges::range_value_t<Range>, bhalf_t>)
+                  !std::is_same_v<ranges::range_value_t<Range>, bhalf_t> &&
+                  !std::is_same_v<ranges::range_value_t<Range>, f8_t> &&
+                  !std::is_same_v<ranges::range_value_t<Range>, bf8_t>)
 #ifdef CK_EXPERIMENTAL_BIT_INT_EXTENSION_INT4
                      || std::is_same_v<ranges::range_value_t<Range>, int4_t>
 #endif

